src/machine/machine_atmega.go

```diff
@@ -5,11 +5,120 @@ package machine
 
 import (
 	"device/avr"
+	"errors"
 	"runtime/interrupt"
 	"runtime/volatile"
 	"unsafe"
 )
 
+// EECR bitfield values.
+// The constants provided by the AVR package are not correct, so these are the correct fields instead.
+const (
+	eecrEERE  = avr.EECR_EERE - 1
+	eecrEEPE  = avr.EECR_EEPE - 1
+	eecrEEMPE = avr.EECR_EEMPE - 2
+	eecrEERIE = avr.EECR_EERIE - 5
+	eecrEEPM0 = avr.EECR_EEPM0 - 12
+	eecrEEPM1 = avr.EECR_EEPM1 - 27
+)
```

Comment:
The bitfields are correct. You have to use them as bitfields, not as bit indices.